src/envs/pybullet_robots.py

```diff
@@ -138,29 +138,36 @@ def get_state(self) -> Array:
         raise NotImplementedError("Override me!")
 
     @abc.abstractmethod
-    def get_joints(self) -> Sequence[float]:
-        """Get the joint states from the current PyBullet state."""
+    def get_joints(self) -> JointsState:
+        """Get the joints state from the current PyBullet state."""
         raise NotImplementedError("Override me!")
 
     @abc.abstractmethod
-    def set_motors(self, action_arr: Array) -> None:
-        """Update the motors to execute the given action in PyBullet.
-
-        The action_arr is an array of desired arm joint values.
-        """
+    def set_motors(self, joints_state: JointsState) -> None:
+        """Update the motors to move toward the given joints state."""
         raise NotImplementedError("Override me!")
 
     @abc.abstractmethod
-    def forward_kinematics(self, action_arr: Array) -> Pose3D:
-        """Compute the end effector pose that would result from executing the
-        given action in PyBullet. The action_arr is an array of desired arm
-        joint values.
+    def forward_kinematics(self, joints_state: JointsState) -> Pose3D:
+        """Compute the end effector pose that if the robot arm joints state
+        was equal to the input joints_state.
 
         WARNING: This method will make use of resetJointState(), and so it
         should NOT be used during simulation.
         """
         raise NotImplementedError("Override me!")
 
+    @abc.abstractmethod
+    def inverse_kinematics(self, end_effector_pose: Pose3D,
+                           validate: bool) -> JointsState:
+        """Compute a joints state from an end effector pose.
+
+        If validate is True, guarantee that the returned joints state
```

Comment:
add a word of caution about validate=True overriding physics and so it shouldn't be used with stepsimulation